client/src/main/java/com/scalar/dl/client/service/GenericContractClientService.java

```diff
@@ -434,6 +444,301 @@ public LedgerValidationResult validateLedger(
+   * @throws ClientException if a request fails for some reason
+   */
+  public LedgerValidationResult validateRecord(
+      String tableName, String columnName, ValueNode value) {
+    return clientService.validateLedger(
+        buildAssetId(
+            AssetType.RECORD, ImmutableList.of(tableName, columnName, toStringFrom(value))));
+  }
+
+  /**
+   * Validates the specified record in the ledger.
+   *
+   * @param tableName a table name
+   * @param columnName a primary key column name
+   * @param value a primary key column value
+   * @param startAge an age to be validated from (inclusive)
+   * @param endAge an age to be validated to (inclusive)
+   * @return {@link LedgerValidationResult}
+   * @throws ClientException if a request fails for some reason
+   */
+  public LedgerValidationResult validateRecord(
+      String tableName, String columnName, ValueNode value, int startAge, int endAge) {
+    return clientService.validateLedger(
+        buildAssetId(
+            AssetType.RECORD, ImmutableList.of(tableName, columnName, toStringFrom(value))),
+        startAge,
+        endAge);
+  }
+
+  /**
+   * Validates the specified record in the ledger.
+   *
+   * @param tableName a table name
+   * @param columnName a primary key column name
+   * @param value a primary key column value in a JSON format
+   * @return {@link LedgerValidationResult}
+   * @throws ClientException if a request fails for some reason
+   */
+  public LedgerValidationResult validateRecord(String tableName, String columnName, String value) {
+    return clientService.validateLedger(
+        buildAssetId(
+            AssetType.RECORD,
+            ImmutableList.of(
+                tableName, columnName, toStringFrom(jacksonSerDe.deserialize(value)))));
+  }
+
+  /**
+   * Validates the specified record in the ledger.
+   *
+   * @param tableName a table name
+   * @param columnName a primary key column name
+   * @param value a primary key column value in a JSON format
+   * @param startAge an age to be validated from (inclusive)
+   * @param endAge an age to be validated to (inclusive)
+   * @return {@link LedgerValidationResult}
+   * @throws ClientException if a request fails for some reason
+   */
+  public LedgerValidationResult validateRecord(
+      String tableName, String columnName, String value, int startAge, int endAge) {
+    return clientService.validateLedger(
+        buildAssetId(
+            AssetType.RECORD,
+            ImmutableList.of(tableName, columnName, toStringFrom(jacksonSerDe.deserialize(value)))),
+        startAge,
+        endAge);
+  }
+
+  /**
+   * Validates the specified index entry in the ledger.
+   *
+   * @param tableName a table name
+   * @param columnName an index key column name
+   * @param value an index key column value
+   * @return {@link LedgerValidationResult}
+   * @throws ClientException if a request fails for some reason
+   */
+  public LedgerValidationResult validateIndexEntry(
```

Comment:
```suggestion
  public LedgerValidationResult validateIndexRecord(
```
Sorry, I suggested `Index Entry`, but `Index Record` might be better and more consistent.
What do you think?

Reply:
@feeblefakie Right, this basically validates an index record (asset) that consists of several index entries. Fixed in 1ee6da9. PTAL again!

Note: we still use "index entry/entries" in several places of Scan/Update contracts, but I think they are consistent since the index record (asset) = index entries = multiple "index entry".